Crash in AndroidChatServiceResponder

```
java.lang.RuntimeException: Error receiving broadcast Intent { act=de.qabel.chat.read flg=0x10 pkg=de.qabel.qabel } in de.qabel.qabelbox.chat.services.AndroidChatServiceResponder@b50cf8f
	at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:891)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5420)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.os.Bundle.unparcel()' on a null object reference
	at android.os.Bundle.putAll(Bundle.java:182)
	at android.content.Intent.putExtras(Intent.java:6899)
	at de.qabel.qabelbox.chat.services.AndroidChatServiceResponder.onReceive(AndroidChatServiceResponder.kt:25)
	at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:881)
	... 7 more
```

--- a/qabelbox/src/main/java/de/qabel/qabelbox/chat/services/AndroidChatServiceResponder.kt
+++ b/qabelbox/src/main/java/de/qabel/qabelbox/chat/services/AndroidChatServiceResponder.kt
@@ -6,6 +6,7 @@ import android.content.Intent
 import de.qabel.qabelbox.QblBroadcastConstants
 import org.jetbrains.anko.AnkoLogger
 import org.jetbrains.anko.info
+import org.jetbrains.anko.warn
 
 /**
  * Starts the Service on receiving QblBroadcastConstants.Chat.NOTIFY_NEW_MESSAGES
@@ -15,14 +16,20 @@ class AndroidChatServiceResponder : BroadcastReceiver(), AnkoLogger {
 
     override fun onReceive(context: Context, intent: Intent?) {
         info("Receive notify broadcast. Start service")
+        if (intent == null) {
+            warn("AndroidChatServiceResponder called with a null intent")
+            return
+        }
         var action = QblBroadcastConstants.Chat.Service.MESSAGES_READ
-        if (intent?.action == QblBroadcastConstants.Chat.NOTIFY_NEW_MESSAGES) {
+        if (intent.action == QblBroadcastConstants.Chat.NOTIFY_NEW_MESSAGES) {
             action = QblBroadcastConstants.Chat.Service.NOTIFY
             abortBroadcast()
         }
         context.applicationContext.startService(Intent(action,
                 null, context.applicationContext, AndroidChatService::class.java).apply {
-            putExtras(intent?.extras)
+            intent.extras.let {
+                putExtras(it)
+            }
         })
     }
 

--- a/qabelbox/src/main/java/de/qabel/qabelbox/startup/activities/WelcomeScreenActivity.java
+++ b/qabelbox/src/main/java/de/qabel/qabelbox/startup/activities/WelcomeScreenActivity.java
@@ -98,7 +98,9 @@ public class WelcomeScreenActivity extends FragmentActivity implements ViewPager
                 if (mCurrentPage < NUM_PAGES - 1) {
                     mPager.setCurrentItem(NUM_PAGES - 1);
                 } else {
-                    if (((WelcomeDisclaimerFragment) fragments[mCurrentPage]).getCheckedState()) {
+                    Fragment fragment = fragments[mCurrentPage];
+                    if (fragment == null ||
+                            ((WelcomeDisclaimerFragment) fragment).getCheckedState()) {
                         finish();
                         prefs.setWelcomeScreenShownAt(System.currentTimeMillis());
                         Intent intent = new Intent(self, CreateAccountActivity.class);

--- a/qabelbox/src/main/java/de/qabel/qabelbox/startup/fragments/CreateAccountResetPasswordFragment.java
+++ b/qabelbox/src/main/java/de/qabel/qabelbox/startup/fragments/CreateAccountResetPasswordFragment.java
@@ -105,8 +105,15 @@ public class CreateAccountResetPasswordFragment extends BaseIdentityFragment {
 
             @Override
             protected void onError(Exception e, @Nullable Response response) {
-                dialog.dismiss();
-                Toast.makeText(getActivity(), R.string.server_access_failed_or_invalid_check_internet_connection, Toast.LENGTH_LONG).show();
+                getActivity().runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (dialog != null) {
+                            dialog.dismiss();
+                        }
+                        Toast.makeText(getActivity(), R.string.server_access_failed_or_invalid_check_internet_connection, Toast.LENGTH_LONG).show();
+                    }
+                });
             }
 
             @Override
@@ -122,9 +129,14 @@ public class CreateAccountResetPasswordFragment extends BaseIdentityFragment {
                         }
                     });
                 } else {
-                    String text = generateErrorMessage(result);
-                    dialog.dismiss();
-                    UIHelper.showDialogMessage(getActivity(), R.string.dialog_headline_info, text);
+                    getActivity().runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            String text = generateErrorMessage(result);
+                            dialog.dismiss();
+                            UIHelper.showDialogMessage(getActivity(), R.string.dialog_headline_info, text);
+                        }
+                    });
                 }
             }
 

